src/components/NavBar/NavBar.component.jsx

```diff
@@ -1,33 +1,40 @@
 import React, { useState } from 'react';
+import { useGlobal } from '../../providers/Global';
 import { API_BASE_URL } from '../../utils/constants';
+import { useFetch } from '../../utils/hooks/useFetch';
 import { NavBar as Navigation, SearchFieldWrapper, SearchField, CheckBox, CheckBoxLabel, CheckBoxWrapper, LoginButton, RightDiv } from './styled';
 
-const NavBar = ({ onVideosChanged }) => {
-    const [searchValue, setSearchValue] = useState('elon musk on TED Talks');
-    const [darkMode, setDarkMode] = useState(true);
+const NavBar = () => {
+    const { state, dispatch } = useGlobal();
+    const { searchKeyword, isDarkMode } = state;
+    const [darkMode, setDarkMode] = useState(isDarkMode);
+    const [url, setUrl] = useState('');
+    useFetch(url, 'updateVideos');
 
     async function searchVideos(keystroke) {
         if (keystroke === 'Enter') {
-            let response = await fetch(`${API_BASE_URL}?part=snippet&maxResults=25&q=${searchValue}&key=${process.env.REACT_APP_API_KEY}`);
-
-            if (response.ok) { // if HTTP-status is 200-299
-                let json = await response.json();
-                onVideosChanged(json.items);
-            } else {
-                console.log(`HTTP-Error: ${response.status}`);
-            }
+            setUrl(`${API_BASE_URL}?part=snippet&maxResults=25&q=${encodeURI(searchKeyword)}&key=${process.env.REACT_APP_API_KEY}`);
         }
     }
 
+    function updateKeyword(e) {
+        return dispatch({ type: 'search', payload: e.target.value });
+    }
+
+    function toggleTheme() {
+        setDarkMode(!darkMode);
+        dispatch({ type: 'theme', payload: !darkMode });
+    }
+
     return (
         <Navigation>
             <SearchFieldWrapper>
                 <span><i className="fa fa-search"></i></span>
-                <SearchField placeholder="search..." value={searchValue} onChange={(e) => setSearchValue(e.target.value)} onKeyUp={(e) => searchVideos(e.key)}></SearchField>
+                <SearchField placeholder="search..." value={searchKeyword} onChange={(e) => updateKeyword(e)} onKeyUp={(e) => searchVideos(e.key)}></SearchField>
             </SearchFieldWrapper>
             <RightDiv>
                 <CheckBoxWrapper >
-                    <CheckBox id="checkbox" type="checkbox" checked={darkMode} onChange={() => setDarkMode(!darkMode)} />
+                    <CheckBox id="checkbox" type="checkbox" checked={darkMode} onChange={() => toggleTheme()} />
```

Comment:
if you drop the local state this should be something like
```suggestion
                    <CheckBox id="checkbox" type="checkbox" checked={isDarkMode} onChange={toggleTheme} />
```

Also I see you have `onChange={() => toggleTheme()}` but since `toggleTheme` is the only function you are executing then you can just pass it as prop to `onChange` like this:` onChange={toggleTheme}`